frontend/packages/console-dynamic-plugin-sdk/src/app/AppInitSDK.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import useReduxStore from './useReduxStore';
+
+type AppInitSDKProps = {
+  children: React.ReactNode;
+};
+
+/**
+ * Component for providing store access to the SDK.
+ * Add this at app-level to make use of app store, preferred to have it under Provider.
+ * It checks for store instance if present or not.
+ * If the store is there then the reference is persisted to be used in SDK else it creates a new store and passes it to the children with the provider
+ * @component AppInitSDK
+ * @example
+ * ```ts
+ * return (
+ *  <Provider store={store}>
+ *   <AppInitSDK>
+ *      <CustomApp />
+ *      ...
+ *   </AppInitSDK>
+ *  </Provider>
+ * )
+ * ```
+ */
+const AppInitSDK: React.FC<AppInitSDKProps> = ({ children }) => {
```

Comment:
Minor nit - we may want to make this name even more explicit as it will help in plugin development when React throws errors and if we introduce error boundaries. Maybe `DynamicPluginSDKAppInit`? @andrewballantyne opinions?

Reply:
later this will be part of the core SDK once we move it out to a separate repo, so was not sure and just called it SDK.

Reply:
Hmmm... the Apps will use it to init the SDK... but we are called the DynamicPluginSDK 🤔 

I think it's fine the way it is, the component may look a little unqualified in its usage... but the full context of what SDK it's initing will be declared in the import definition. eg:

```
import { AppInitSDK } from '@openshift/dynamic-plugin-sdk';
```